### PR TITLE
Update Main Code

### PR DESCRIPTION
**Pull request by ifenta**
Working random dropping blocks, speed control, block movement, and stopping on block

### CODE REVIEW

File: src/block.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def __init__(self, color, width, height, position):
         self.width = width
         self.height = height
         self.position = position
+        self.alive = False
```
Comment (cinabars) [inline src/block.py]: `self.occupied`

File: src/grid.py [hunk 2]
```diff
@@ -6,18 +6,97 @@ class Grid:
     """
     def __init__(self, width):
 
-        if width*2 > ((width*2) + int(width/10)*4):
-            raise IndexError('height must be twice of width')
```
Comment (cinabars) [inline src/grid.py:10]: :)

File: src/grid.py [hunk 3]
```diff
@@ -6,18 +6,97 @@ class Grid:
...
         self.width = width
         self.height = ((width*2) + int(width/10)*4)
+        self.GRID_WIDTH = 10
+        self.GRID_HEIGHT = 24
+        self.block_width = self.width/self.GRID_WIDTH
+        self.block_height = self.height/self.GRID_HEIGHT
 
         self.grid = []
         
         for ii in range(self.width):
```
Comment (cinabars) [inline src/grid.py]: IMO `x`/`y` are better than `ii`/`jj`

File: src/grid.py [hunk 4]
```diff
@@ -6,18 +6,97 @@ class Grid:
...
             self.grid.append([])
             for jj in range(self.height):
-                self.grid[ii].append(block.Block((0,0,0),self.width/10,self.height/24,(ii*(self.width/10),jj*(self.height/24))))
+                self.grid[ii].append(block.Block((0,0,0),self.block_width,self.block_height,(ii*(self.block_width),jj*(self.block_height))))
+
+    def move_down(self, screen, pos):
+        if pos != None:
+            if pos[1] < self.GRID_HEIGHT-1:
```
Comment (cinabars) [inline src/grid.py]: combine these two lines into one `if` block: `if pos != None and pos[1] < self.GRID_HEIGHT - 1:`

File: src/grid.py [hunk 5]
```diff
@@ -6,18 +6,97 @@ class Grid:
...
+                old_blk = self.grid[pos[0]][pos[1]]
```
Comment (cinabars) [inline src/grid.py]: why not just spell out `block` and `position`

File: src/tetris.py [hunk 6]
```diff
@@ -19,37 +17,89 @@
...
 screen.fill((0,0,0))
 
-
+'''
 # Sets each block in the grid to a random color
 for x in grid.grid:
 	for y in x:
 		y.surf.fill( ( random.randint(0,255),random.randint(0,255),random.randint(0,255) ) )  
 		screen.blit(y.surf, y.position)
+'''
```
Comment (cinabars) [inline src/tetris.py]: just delete this

File: src/tetris.py [hunk 7]
```diff
@@ -19,37 +17,89 @@
...
-		
-		# Check for KEYDOWN event 
-		if event.type == KEYDOWN: 
+pos = [4,0]
+
+blk = grid.grid[pos[0]][pos[1]]
+
+blk.color = (0,200,255)
+blk.surf.fill( (0,200,255) )  
+screen.blit( blk.surf, blk.position )
```
Comment (cinabars) [inline src/tetris.py]: what does lines 36-42 do?

File: src/tetris.py [hunk 8]
```diff
@@ -19,37 +17,89 @@
...
+				if event.key == K_LEFT:
+					pos = grid.move_left(screen, pos)
+					pygame.display.flip()
+
+				elif event.key == K_RIGHT:
+					pos = grid.move_right(screen, pos)
+					pygame.display.flip()
+
+				elif event.key == K_DOWN:
+					drop_speed -= 100
```
Comment (cinabars) [inline src/tetris.py]: later we will want to make sure the drop speed gets reset back to normal when a new shape comes down.

Comment (cinabars) [reply src/tetris.py]: so the lower `drop_speed` is, the faster it drops? confusing variable name. maybe `drop_delay`

**Review by cinabars — CHANGES_REQUESTED** [review]

File: src/grid.py (hunk 5 above)
Comment (ifenta) [reply src/grid.py]: Fucking piccky

File: src/tetris.py (hunk 7 above)
Comment (ifenta) [reply src/tetris.py]: Its fine. Ill comment it

**Review by cinabars — APPROVED** [review]

**Pull request merged**